src/routes/user.route.ts

```diff
@@ -0,0 +1,146 @@
+ */
+
+/**
+ * @swagger
+ * definitions:
+ *   User:
+ *     type: object
+ *     properties:
+ *       id:
+ *         type: integer
+ *         description: Unique identifier representing a specific user
+ *         example: 2
+ *       name:
+ *         type: string
+ *         description: name of the user
+ *         example: Krishna
+ *       email:
+ *         type: string
+ *         description: email of the user
+ *         required: true
+ *         example: test@gmail.com
+ *       password:
+ *         type: string
+ *         description: password of the user
+ *         required: true
+ *         example: "1234"
+ *   Error:
+ *     type: object
+ *     properties:
+ *        message:
+ *           type: string
+ *        error:
+ *           type: boolean
+ *           default: true
+ */
+
+/**
+ * @swagger
+ * /users:
+ *   post:
+ *     tags: ['User']
+ *     summary: Create a new user
+ *     description: Create a new user
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               name:
+ *                 type: string
+ *                 description: User's name
+ *                 example: Toto
+ *               email:
+ *                 type: string
+ *                 description: User's email
+ *                 example: Toto@gmail.com
+ *               password:
+ *                 type: string
+ *                 description: User's password
+ *                 example: azerty
+ *     responses:
+ *       201:
+ *         description: The created user
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/User'
+ */
+router.post('/', UserController.create)
+
+/**
+ * @swagger
+ * /users/{id}:
+ *   put:
+ *     tags: ['User']
+ *     summary: Update an user
+ *     description: Update an user
+ *     parameters:
+ *       - id: id
+ *         in: path
+ *         description: User id
+ *         required: true
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               name:
+ *                 type: string
+ *                 description: User's name
+ *                 example: Tutu
+ *               email:
+ *                 type: string
+ *                 description: User's email
+ *                 example: Tutu@gmail.com
+ *               password:
+ *                 type: string
+ *                 description: User's password
+ *                 example: poiuyt
+ *     responses:
+ *       200:
+ *         description: The updated user
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/User'
+ *       404:
+ *         description: User not found
+ */
+router.put('/:id', UserController.update)
```

Comment:
by applying authenticate middleware, we won't need to pass the user id in the route
(user can only change their own information)

```suggestion
router.put('', UserController.update)
```